core/src/components/datetime/utils/state.ts

```diff
@@ -102,3 +142,81 @@ export const getCalendarDayState = (
     ariaLabel: generateDayAriaLabel(locale, isToday, refParts)
   }
 }
+
+/**
+ * Returns `true` if the month is disabled given the
+ * current date value and min/max date constraints.
+ */
+export const isMonthDisabled = (refParts: { month: number, year: number }, { minParts, maxParts }: {
+  minParts?: { month?: number, year: number },
+  maxParts?: {
+    month?: number, year: number
+  }
+}) => {
+  // If the min date is set and the year is less than the min year.
+  if (minParts && minParts.year > refParts.year) {
+    return true;
+  }
+  // If the max date is set and the year is greater than the max year.
+  if (maxParts && maxParts.year < refParts.year) {
+    return true;
+  }
+  // If the min date is set and the year is the same as the min year,
+  // but the month is less than the min month.
+  if (minParts && minParts.year === refParts.year && minParts.month !== undefined && minParts.month > refParts.month) {
+    return true;
+  }
+  // If the max date is set and the year is the same as the max year,
+  // but the month is greater than the max month.
+  if (maxParts && maxParts.year === refParts.year && maxParts.month !== undefined && maxParts.month < refParts.month) {
+    return true;
+  }
+
+  return false;
+}
+
+/**
+ * Given a working date, an optional minimum date range,
+ * and an optional maximum date range; determine if the
+ * previous navigation button is disabled.
+ */
+export const isPrevMonthDisabled = (
+  refParts: {
+    month: number,
+    year: number,
+    day: number | null
+  },
+  minParts?: {
+    month?: number,
+    year: number
+  },
+  maxParts?: {
+    month?: number,
+    year: number
+  }) => {
```

Comment:
We do we use this instead of `DatetimeParts`?

Reply:
The `DatetimeParts` type is a bit of a beast. It's being used for a lot of utility functions that only require the `month` or `year`. As a result, we are actually overriding Typescript's ability to do type inference and using the postfix `!` operator to remove the `null` & `undefined` type. 

Really these functions should be as open as possible, only requiring the date parts that is uses or is dependent on based on inner function calls. 

I can refactor these to use `DatetimeParts`, but it makes the API usage something like:
```ts
isNextMonthDisabled({ month: 10, year: 2021, day: undefined })
```

Reply:
Ahh I see. Ok let's keep it as you have it 👍

Reply:
I could see it making sense being consistent with `Datetimeparts` for the type here, but re-evaluating across the implementation if simpler types could exist for the utils. 

If you're good with that, I can create a tech debt task.

Reply:
Yeah I see your point about `DatetimeParts` being excessive for something like this. Tech debt to explore a better way to do this would be good.